File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.cpp
class Solution {
public:
    string longestPalindrome(string s) {

        string palindrome = "";

        for (int left = 0; left < s.length(); left++) {
            int right = s.size() - 1;

            while (right >= left) {
                if (s[left] == s[right] && right - left + 1 > palindrome.length() && isPalindrome(s, left, right)) {
                    palindrome = s.substr(left, right - left + 1);
                }
                right--;
            }
        }

        return palindrome;
        
    }

private:
    bool isPalindrome(string s, int l, int r) {
        while (l < r) {
            if (s[l] != s[r]) return false;
            l++;
            r--;
        }

        return true;
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        output = []

        if not root:
            return output

        q = deque()
        q.append(root)

        while q:
            levelNodes = []
            levelSize = len(q)
            for i in range(levelSize):
                node = q.popleft()
                levelNodes.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            
            output.append(levelNodes)
        
        return output